src/Microsoft.Health.Fhir.SqlServer/Features/Storage/Registry/SqlServerStatusRegistry.cs

```diff
@@ -0,0 +1,115 @@
+            _updateSearchParamRegistryTvpGenerator = updateSearchParamRegistryTvpGenerator;
+            _insertSearchParamRegistryTvpGenerator = insertSearchParamRegistryTvpGenerator;
+        }
+
+        public async Task<IReadOnlyCollection<ResourceSearchParameterStatus>> GetSearchParameterStatuses(CancellationToken cancellationToken)
+        {
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                VLatest.GetSearchParamStatuses.PopulateCommand(sqlCommandWrapper);
+
+                var parameterStatuses = new List<ResourceSearchParameterStatus>();
+
+                using (SqlDataReader sqlDataReader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                {
+                    while (await sqlDataReader.ReadAsync())
+                    {
+                        (string uri, string stringStatus, DateTimeOffset? lastUpdated, bool isPartiallySupported) = sqlDataReader.ReadRow(
+                            VLatest.SearchParamRegistry.Uri,
+                            VLatest.SearchParamRegistry.Status,
+                            VLatest.SearchParamRegistry.LastUpdated,
+                            VLatest.SearchParamRegistry.IsPartiallySupported);
+
+                        var status = Enum.Parse<SearchParameterStatus>(stringStatus, true);
+
+                        var resourceSearchParameterStatus = new ResourceSearchParameterStatus()
+                        {
+                            Uri = new Uri(uri),
+                            Status = status,
+                            IsPartiallySupported = isPartiallySupported,
+                            LastUpdated = (DateTimeOffset)lastUpdated,
+                        };
+
+                        parameterStatuses.Add(resourceSearchParameterStatus);
+                    }
+                }
+
+                return parameterStatuses;
+            }
+        }
+
+        public async Task UpsertStatuses(IEnumerable<ResourceSearchParameterStatus> statuses, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(statuses, nameof(statuses));
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                VLatest.UpsertSearchParamStatus.PopulateCommand(sqlCommandWrapper, _updateSearchParamRegistryTvpGenerator.Generate(statuses.ToList()));
+
+                await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
+            }
+        }
+
+        internal async Task<bool> GetIsSearchParameterRegistryEmpty(CancellationToken cancellationToken)
```

Comment:
nit: GetIsSearchParameterRegistryEmpty -> IsSearchParameterRegistryEmpty

Reply:
Thanks, fixed in e6b9547.